component/async/sqs/option.go

```diff
@@ -13,53 +13,49 @@ type OptionFunc func(*Factory) error
 
 // MaxMessages option for setting the max number of messages fetched.
 // Allowed values are between 1 and 10.
+// If this option is unused, it defaults to 3.
+// If messages can be processed very quickly, maxing out this value is fine, otherwise having a high value is risky as it might trigger the visibility timeout.
+// Having a value too small isn't recommended either, as it increases the number of SQS API requests, thus AWS costs.
 func MaxMessages(maxMessages int64) OptionFunc {
 	return func(f *Factory) error {
 		if maxMessages <= 0 || maxMessages > 10 {
 			return errors.New("max messages should be between 1 and 10")
 		}
-		f.maxMessages = maxMessages
+		f.maxMessages = &maxMessages
```

Comment:
Any reason why this has to be a pointer?

Reply:
as mentioned earlier, helps with just delegating default values to the queue